lib/node_modules/@stdlib/repl/lib/main.js

```diff
@@ -782,6 +800,191 @@ setNonEnumerableReadOnly( REPL.prototype, 'clearUserDocs', function clearUserDoc
 	return this;
 });
 
+/**
+* Gets all available themes in the syntax-highlighter.
+*
+* @name themes
+* @memberof REPL.prototype
+* @type {Function}
+* @returns {Array} array of all theme names
+*
+* @example
+* var debug = require( '@stdlib/streams/node/debug-sink' );
+*
+* // Create a new REPL:
+* var repl = new REPL({
+*     'output': debug()
+* });
+*
+* // ...
+*
+* // Fetch all available themes:
+* repl.themes();
+*
+* // ...
+*
+* // Close the REPL:
+* repl.close();
+*/
+setNonEnumerableReadOnly( REPL.prototype, 'themes', function themes() {
+	return this._syntaxHighlighter.getThemes();
+});
+
+/**
+* Gets a theme's color palette from the syntax-highlighter.
+*
+* @name getTheme
+* @memberof REPL.prototype
+* @type {Function}
+* @param {string} theme - theme name
+* @throws {TypeError} argument must be a string
+* @returns {Object} theme object
+*
+* @example
+* var debug = require( '@stdlib/streams/node/debug-sink' );
+*
+* // Create a new REPL:
+* var repl = new REPL({
+*     'output': debug()
+* });
+*
+* // ...
+*
+* // Get theme object:
+* repl.getTheme( 'myTheme' );
+*
+* // ...
+*
+* // Close the REPL:
+* repl.close();
+*/
+setNonEnumerableReadOnly( REPL.prototype, 'getTheme', function getTheme( theme ) {
+	if ( !isString( theme ) ) {
+		throw new TypeError( format( 'invalid argument. First argument must be a string. Value: `%s`.', theme ) );
+	}
+	return this._syntaxHighlighter.getThemeConfig( theme );
+});
+
+/**
+* Adds a theme to the syntax-highlighter.
+*
+* @name addTheme
+* @memberof REPL.prototype
+* @type {Function}
+* @param {string} name - theme name
+* @param {Object} theme - theme object
+* @throws {TypeError} first argument must be a string
+* @throws {TypeError} second argument must be an object
+* @returns {void}
+*
+* @example
+* var debug = require( '@stdlib/streams/node/debug-sink' );
+*
+* // Create a new REPL:
+* var repl = new REPL({
+*     'output': debug()
+* });
+*
+* // ...
+*
+* // Add a user-defined theme:
+* repl.addTheme( 'myTheme', {
+*     'keyword': 'red',
+*     'variable': 'green'
+*
+*     // ...
+* });
+*
+* // ...
+*
+* // Close the REPL:
+* repl.close();
+*/
+setNonEnumerableReadOnly( REPL.prototype, 'addTheme', function addTheme( name, theme ) {
+	if ( !isString( name ) ) {
+		throw new TypeError( format( 'invalid argument. First argument must be a string. Value: `%s`.', name ) );
+	}
+	if ( !isPlainObject( theme ) ) {
```

Comment:
Would be good to add schema and validation for the theme object. Can be reserved for a follow-up PR.